apps/src/code-studio/initApp/project.js

```diff
@@ -460,9 +464,12 @@ var projects = module.exports = {
       callback = args[0];
       forceNewVersion = args[1];
       this.sourceHandler.getAnimationList(animations => {
-        const source = this.sourceHandler.getLevelSource();
-        const html = this.sourceHandler.getLevelHtml();
-        this.save({source, html, animations}, callback, forceNewVersion);
+        var thisProject = this;
+        this.sourceHandler.getLevelSource().then(function (response) {
```

Comment:
this could be an arrow function, which also removes the need to capture `this` as `thisProject`:
```
this.sourceHandler.getLevelSource().then(response => {
  const source = response;
  const html = thisProject.sourceHandler.getLevelHtml();
  this.save({source, html, animations}, callback, forceNewVersion);
})